app/services/users_service.rb

```diff
@@ -4,7 +4,14 @@ class UsersService < BaseService
   def login(email, password)
     result.user = User.find_by(email: email)&.authenticate(password)
     result.token = generate_token if result.user
-    result.fail!('incorrect_login_or_password') unless result.user
+
+    unless result.user
+      result.fail!('incorrect_login_or_password')
+      return result
+    end
```

Comment:
Small tip: `fail!` method is returning the result object so that you can do:

```suggestion
    return result.fali!('incorrect_login_or_password') unless result.user
```

Reply:
Nice, thanks 👍